orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/poller/join/MinionAcceptor.java

```diff
@@ -43,15 +44,21 @@ public void acceptMinions() throws CloudbreakOrchestratorFailedException {
             MinionKeysOnMasterResponse minionKeysOnMaster = fetchMinionsFromMaster(sc, minions);
             List<String> unacceptedMinions = minionKeysOnMaster.getUnacceptedMinions();
             List<String> deniedMinions = minionKeysOnMaster.getDeniedMinions();
-            removeMinionIdsInBothDeniedAndUnacceptedState(sc, deniedMinions, unacceptedMinions);
+            cleanupMinionIds(sc, deniedMinions, unacceptedMinions);
             if (!unacceptedMinions.isEmpty()) {
                 proceedWithAcceptingMinions(sc, unacceptedMinions);
             } else {
-                LOGGER.info("No unaccepted minions found on master");
+                LOGGER.info("No unaccepted minions found on master: [{}]", sc.getHostname());
             }
         }
     }
 
+    private void cleanupMinionIds(SaltConnector sc, List<String> deniedMinions, List<String> unacceptedMinions)
+            throws CloudbreakOrchestratorFailedException {
+        removeMinionIdsInBothDeniedAndUnacceptedState(sc, deniedMinions, unacceptedMinions);
```

Comment:
As I see with this change we remove every denied minions (first step is delete denied and unaccepted, then delete other denied). So I think it would be better to delete every denied minion in one step. Code will be easier to understand and it will do only one request to salt master. You can throw exceptionin this case as it was thrown in unaccepted+denied case, something like this: 

new CloudbreakOrchestratorFailedException("There were minions in denied state: " + deniedMinions);

Reply:
Why would I need to throw an exception?

Ok, I can build a list of (denied && unaccepted) || denied

Reply:
So upon further thinking, each salt call is fixing a different problem.  So maybe we should leave as is.  If we run into a scenario where we have both being executed and it doesn't fix the issue, it'd be good to log what it was doing rather than a single large call with less logging.

Reply:
If you remove the dead minions then "target" nodes' keys will not be accepted in this iteration. It will not cause any trouble, but we will do the minion ip addr collection, but it will fail for sure because the former denied minions will be missing. So this was the reason I thrown an exception when I removed minions both in unaccepted and denied state. But I don't push it, with this we can skip an unnecessary call.

Reply:
Ok, we can leave it as is, it is an edge case so we should not remove any minions in normal circumstances.